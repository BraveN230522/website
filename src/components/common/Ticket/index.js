import Router from 'next/router';
import React from 'react';
import { sendEvent } from '@utils/google-services/gtm';

import {
  TicketStyles,
  TicketFigure,
  TicketSubtitle,
  TicketTitle,
  TicketImage,
  TicketLegend,
  TicketPriceContainer,
  TicketAverage,
} from './styles';

const Ticket = ({ dark, link, image, title, subtitle, price, promo, className, hidePrice }) => {
  const handleTicketClicked = e => {
    e.preventDefault();

    sendEvent('featuredClick', {
      featured: {
        detail: {
          name: title,
          link,
          promo,
          price,
          image,
          subtitle,
        },
      },
    });
    window.location.href = link;

    // Router.push({
    //   pathname: link,
    // });
  };

  return (
    <TicketStyles className={className}>
      <a onClick={handleTicketClicked}>
        <TicketFigure>
          <TicketImage src={image} alt={title} />

          {/* {promo && (
            <TicketLegend>
              <span>{promo}</span>
            </TicketLegend>
          )} */}
        </TicketFigure>
      </a>

      <a onClick={handleTicketClicked}>
        <TicketTitle dark={dark}>{title}</TicketTitle>
        <TicketSubtitle dark={dark}>{subtitle}</TicketSubtitle>
        {!hidePrice && (
          <TicketPriceContainer dark={dark}>
            From <span>${price || 'NA'}</span>
            {/* From <span>${price}</span> <TicketAverage>{average}</TicketAverage> */}
          </TicketPriceContainer>
        )}
      </a>
    </TicketStyles>
  );
};

export default Ticket;
