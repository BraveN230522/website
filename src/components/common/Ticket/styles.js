import styled from '@emotion/styled';

const TicketStyles = styled.div`
  max-width: 225px;
  width: 100%;

  a {
    text-decoration: none;
  }
`;

const TicketFigure = styled.figure`
  position: relative;
  margin: 0;

  &:hover figcaption {
    box-shadow: inset -60px 0 0px ${({ theme }) => theme.colors.white};
    color: ${({ theme }) => theme.colors.swiamBlue};
    transform: scale(0.9);
  }
`;

const TicketImage = styled.img`
  width: 100%;
  height: 137px;
  object-fit: cover;
  border-radius: 8px;
`;

const TicketLegend = styled.figcaption`
  position: absolute;
  top: 6px;
  right: 11px;
  width: 57px;
  height: 55px;
  border-radius: 27.6px;
  background: ${({ theme }) => theme.colors.swiamBlue};
  color: ${({ theme }) => theme.colors.white};
  display: flex;
  align-items: center;
  font-family: ${({ theme }) => theme.fonts.Helvetica};
  font-size: 12.4px;
  letter-spacing: -0.09px;
  flex-direction: column;
  justify-content: center;
  transition: all 0.2s cubic-bezier(1, -0.45, 1, 1), color 0.2s 0.3s ease-in, transform 0.3s ease-out;
  box-shadow: inset 0 0 0px ${({ theme }) => theme.colors.white};
  text-align: center;

  span {
    font-size: 14.4px;
    letter-spacing: -0.13px;
  }
`;

const TicketTitle = styled.p`
  margin: 0;
  font-size: 16px;
  letter-spacing: -0.11px;
  color: ${({ dark, theme }) => (dark ? theme.colors.white : theme.colors.black)};
  margin-top: 8px;
  opacity: 0.8;
`;

const TicketSubtitle = styled.p`
  color: ${({ dark, theme }) => (dark ? theme.colors.white : theme.colors.lightGrey)};
  font-size: 12px;
  font-weight: 500;
  opacity: 0.8;
  letter-spacing: -0.1px;
  margin: 0;
  margin-bottom: 8px;
`;

const TicketPriceContainer = styled.div`
  color: ${({ dark, theme }) => (dark ? theme.colors.white : theme.colors.normalGrey)};
  display: flex;
  font-size: 12px;
  align-items: center;
  letter-spacing: -0.09px;
  line-height: 17px;

  span {
    font-weight: 600;
  }

  span,
  p {
    margin-left: 6px;
  }
`;

const TicketAverage = styled.p`
  color: ${({ theme }) => theme.colors.white};
  background: ${({ theme }) => theme.colors.positive};
  width: 27px;
  height: 17px;
  border-radius: 1.6px;
  margin: 0;
  font-size: 11px;
  font-weight: 500;
  text-align: center;
  line-height: 17px;
`;

export {
  TicketStyles,
  TicketTitle,
  TicketSubtitle,
  TicketAverage,
  TicketPriceContainer,
  TicketFigure,
  TicketLegend,
  TicketImage,
};
