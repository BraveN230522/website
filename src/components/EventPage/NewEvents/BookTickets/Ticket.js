import React from 'react';

import { TicketContainer, TicketItem, TicketDescription, TicketTextContainer, TicketButton } from './styles';

const Ticket = ({ ticket, onPriceClicked }) => {
  return (
    <TicketContainer>
      <TicketTextContainer>
        <TicketItem>{ticket.name}</TicketItem>
        <TicketDescription>
          {ticket.attributes.row !== 'n/a' && `Row ${ticket.attributes.row} - `}
          {ticket.description}
        </TicketDescription>
      </TicketTextContainer>

      <TicketButton onClick={onPriceClicked} outline>
        ${ticket.price.amount} each
      </TicketButton>
    </TicketContainer>
  );
};

export default Ticket;
