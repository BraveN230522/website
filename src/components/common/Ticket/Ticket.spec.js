import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import MCGImage from '@images/mcg-partnership.jpg';
import Ticket from './index';

const ticket = {
  average: 8.5,
  price: '129',
  discount: 20,
  title: 'Hilton Garden Inn',
  subtitle: 'Near Yankee Stadium',
  image: MCGImage,
};

describe('<Ticket />', () => {
  it('Renders the Ticket unchanged', () => {
    const { container } = render(<Ticket {...ticket} />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });

  it('Renders a Ticket without discount', () => {
    const { container } = render(<Ticket {...ticket} discount={null} />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });
});
