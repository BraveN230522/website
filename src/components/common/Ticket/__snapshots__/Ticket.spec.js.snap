// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ticket /> Renders a Ticket without discount 1`] = `
.emotion-12 {
  max-width: 225px;
  width: 100%;
}

.emotion-12 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2 {
  position: relative;
  margin: 0;
}

.emotion-2:hover figcaption {
  box-shadow: inset -60px 0 0px #ffffff;
  color: #2891ff;
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.emotion-0 {
  width: 100%;
  height: 137px;
  object-fit: cover;
  border-radius: 8px;
}

.emotion-4 {
  margin: 0;
  font-size: 16px;
  -webkit-letter-spacing: -0.11px;
  -moz-letter-spacing: -0.11px;
  -ms-letter-spacing: -0.11px;
  letter-spacing: -0.11px;
  color: #000000;
  margin-top: 8px;
  opacity: 0.8;
}

.emotion-6 {
  color: #7c7c7c;
  font-size: 12px;
  font-weight: 500;
  opacity: 0.8;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  margin: 0;
  margin-bottom: 8px;
}

.emotion-10 {
  color: #383838;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-letter-spacing: -0.09px;
  -moz-letter-spacing: -0.09px;
  -ms-letter-spacing: -0.09px;
  letter-spacing: -0.09px;
  line-height: 17px;
}

.emotion-10 span {
  font-weight: 600;
}

.emotion-10 span,
.emotion-10 p {
  margin-left: 6px;
}

.emotion-8 {
  color: #ffffff;
  background: #b1ca46;
  width: 27px;
  height: 17px;
  border-radius: 1.6px;
  margin: 0;
  font-size: 11px;
  font-weight: 500;
  text-align: center;
  line-height: 17px;
}

<div>
  <div
    class="emotion-12 emotion-13"
  >
    <a
      href="#"
    >
      <figure
        class="emotion-2 emotion-3"
      >
        <img
          class="emotion-0 emotion-1"
          src="mocked-file"
        />
      </figure>
    </a>
    <a
      href="#"
    >
      <p
        class="emotion-4 emotion-5"
      >
        Hilton Garden Inn
      </p>
      <p
        class="emotion-6 emotion-7"
      >
        Near Yankee Stadium
      </p>
      <div
        class="emotion-10 emotion-11"
      >
        From 
        <span>
          $
          129
        </span>
         
        <p
          class="emotion-8 emotion-9"
        >
          8.5
        </p>
      </div>
    </a>
  </div>
</div>
`;

exports[`<Ticket /> Renders the Ticket unchanged 1`] = `
.emotion-14 {
  max-width: 225px;
  width: 100%;
}

.emotion-14 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  position: relative;
  margin: 0;
}

.emotion-4:hover figcaption {
  box-shadow: inset -60px 0 0px #ffffff;
  color: #2891ff;
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.emotion-0 {
  width: 100%;
  height: 137px;
  object-fit: cover;
  border-radius: 8px;
}

.emotion-2 {
  position: absolute;
  top: 6px;
  right: 11px;
  width: 57px;
  height: 55px;
  border-radius: 27.6px;
  background: #2891ff;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Helvetica,sans-serif;
  font-size: 12.4px;
  -webkit-letter-spacing: -0.09px;
  -moz-letter-spacing: -0.09px;
  -ms-letter-spacing: -0.09px;
  letter-spacing: -0.09px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.2s cubic-bezier(1,-0.45,1,1),color 0.2s 0.3s ease-in,-webkit-transform 0.3s ease-out;
  -webkit-transition: all 0.2s cubic-bezier(1,-0.45,1,1),color 0.2s 0.3s ease-in,transform 0.3s ease-out;
  transition: all 0.2s cubic-bezier(1,-0.45,1,1),color 0.2s 0.3s ease-in,transform 0.3s ease-out;
  box-shadow: inset 0 0 0px #ffffff;
}

.emotion-2 span {
  font-size: 18.4px;
  -webkit-letter-spacing: -0.13px;
  -moz-letter-spacing: -0.13px;
  -ms-letter-spacing: -0.13px;
  letter-spacing: -0.13px;
}

.emotion-6 {
  margin: 0;
  font-size: 16px;
  -webkit-letter-spacing: -0.11px;
  -moz-letter-spacing: -0.11px;
  -ms-letter-spacing: -0.11px;
  letter-spacing: -0.11px;
  color: #000000;
  margin-top: 8px;
  opacity: 0.8;
}

.emotion-8 {
  color: #7c7c7c;
  font-size: 12px;
  font-weight: 500;
  opacity: 0.8;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  margin: 0;
  margin-bottom: 8px;
}

.emotion-12 {
  color: #383838;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-letter-spacing: -0.09px;
  -moz-letter-spacing: -0.09px;
  -ms-letter-spacing: -0.09px;
  letter-spacing: -0.09px;
  line-height: 17px;
}

.emotion-12 span {
  font-weight: 600;
}

.emotion-12 span,
.emotion-12 p {
  margin-left: 6px;
}

.emotion-10 {
  color: #ffffff;
  background: #b1ca46;
  width: 27px;
  height: 17px;
  border-radius: 1.6px;
  margin: 0;
  font-size: 11px;
  font-weight: 500;
  text-align: center;
  line-height: 17px;
}

<div>
  <div
    class="emotion-14 emotion-15"
  >
    <a
      href="#"
    >
      <figure
        class="emotion-4 emotion-5"
      >
        <img
          class="emotion-0 emotion-1"
          src="mocked-file"
        />
        <figcaption
          class="emotion-2 emotion-3"
        >
          <span>
            20
            %
          </span>
          OFF
        </figcaption>
      </figure>
    </a>
    <a
      href="#"
    >
      <p
        class="emotion-6 emotion-7"
      >
        Hilton Garden Inn
      </p>
      <p
        class="emotion-8 emotion-9"
      >
        Near Yankee Stadium
      </p>
      <div
        class="emotion-12 emotion-13"
      >
        From 
        <span>
          $
          129
        </span>
         
        <p
          class="emotion-10 emotion-11"
        >
          8.5
        </p>
      </div>
    </a>
  </div>
</div>
`;
